### PR TITLE
Add tests for save/load parameters with rowscaling

### LINKED ISSUE #2582: Create test for `analysis::save_row_scaling_parameters` in: update.cpp
Related to: #2574

### LINKED ISSUE #2584: Create test for `analysis::load_row_scaling_parameters` in: update.cpp
Related to: #2574

Note that this is also related to #2582, and a round trip test should probably be created.

### PR DESCRIPTION
**Pull request by ManInFez**
**Issue**
Resolves #2582 
Resolves #2584

**Approach**
Roundtrip test for a matrix with rowscaling attached
## Pre review checklist

- [x] Added appropriate labels

Adding labels helps the maintainers when writing release notes, see sections and the
corresponding labels here: https://github.com/equinor/ert/blob/main/.github/release.yml

### CODE REVIEW

File: libres/tests/analysis/test_save_parameters.cpp [hunk 1]
```diff
@@ -102,3 +116,88 @@ TEST_CASE("Write and read a matrix to enkf_fs instance",
...
+        std::vector<std::pair<matrix_type *, const row_scaling_type *>>
+            row_scaling_list;
+
+        row_scaling_list.push_back(
+            std::pair<matrix_type *, const row_scaling_type *>(A, scaling));
+        analysis::save_row_scaling_parameters(fs, ensemble_config, active_index,
+                                              ministep, row_scaling_list);
+
+        WHEN("loading parameters from enkf_fs") {
+            auto parmeter_matrices = analysis::load_row_scaling_parameters(
```
Comment (dafeda) [inline libres/tests/analysis/test_save_parameters.cpp:185]: `parameter_matrices`? 

Pedantizm 🚀

File: libres/tests/analysis/test_save_parameters.cpp [hunk 2]
```diff
@@ -102,3 +116,88 @@ TEST_CASE("Write and read a matrix to enkf_fs instance",
...
+            THEN("Loading parameters yield the same matrix") {
+                for (int i = 0; i < row_scaling_list.size(); i++) {
+                    auto A = row_scaling_list[i].first;
+                    auto B = row_scaling_list[i].first;
+                    REQUIRE(matrix_equal(A, B));
+                }
+            }
+        }
+
+        //cleanup
```
Comment (dafeda) [inline libres/tests/analysis/test_save_parameters.cpp:196]: I think we should cleanup `scaling` as well with a call to `row_scaling_free`.

File: libres/tests/analysis/test_save_parameters.cpp (hunk 1 above)
Comment (ManInFez) [reply libres/tests/analysis/test_save_parameters.cpp:185]: it is potentially a list of matrices :shrug:

Comment (dafeda) [reply libres/tests/analysis/test_save_parameters.cpp:185]: I was thinking that `parmeter` should be `parameter`...
No biggie 😄

File: libres/tests/analysis/test_save_parameters.cpp (hunk 2 above)
Comment (ManInFez) [reply libres/tests/analysis/test_save_parameters.cpp:196]: The rowscaling object is handled by the ministep object, and destroyed when the ministep is deleted.

File: libres/tests/analysis/test_save_parameters.cpp (hunk 1 above)
Comment (ManInFez) [reply libres/tests/analysis/test_save_parameters.cpp:185]: 🙈  I will fix

**Review by dafeda — APPROVED** [review]
LGTM 👍

**Comment by ManInFez** [comment]
Test this please!

**Comment by ManInFez** [comment]
Test this please!

**Comment by ManInFez** [comment]
Test this please!

**Comment by ManInFez** [comment]
Test this please!

**Pull request merged**